### PR TITLE
feat: configmap, rbac, service, serviceaccount apply annotations-filter

### PR DESCRIPTION
**Pull request by chinaran**
**Link to tracking Issue(s):** <Issue number if applicable>

- Resolves: #3151

### CODE REVIEW

**Review by yuriolisa — APPROVED** [review]

**Comment by jaronoff97** [comment]
@chinaran can you add a change log entry when you get a chance please? Thank you for your contribution!

**Comment by chinaran** [comment]
@jaronoff97 Thanks for the reminder, I've added the changelog.

File: internal/manifests/collector/daemonset_test.go [hunk 1]
```diff
@@ -55,9 +55,6 @@ func TestDaemonSetNewDefault(t *testing.T) {
 	// verify
 	assert.Equal(t, "my-instance-collector", d.Name)
 	assert.Equal(t, "my-instance-collector", d.Labels["app.kubernetes.io/name"])
-	assert.Equal(t, "true", d.Annotations["prometheus.io/scrape"])
-	assert.Equal(t, "8888", d.Annotations["prometheus.io/port"])
-	assert.Equal(t, "/metrics", d.Annotations["prometheus.io/path"])
```
Comment (jaronoff97) [inline internal/manifests/collector/daemonset_test.go:60]: can you add an assertion that this is still on the pod spec?

Comment (jaronoff97) [reply internal/manifests/collector/daemonset_test.go:60]: same for statefulset and deployment

**Review by jaronoff97 — APPROVED** [review]

File: internal/manifests/manifestutils/annotations.go [hunk 2]
```diff
@@ -27,15 +27,6 @@ func Annotations(instance v1beta1.OpenTelemetryCollector, filterAnnotations []st
...
 
-	// Enable Prometheus annotations by default if DisablePrometheusAnnotations is nil or true
-	if !instance.Spec.Observability.Metrics.DisablePrometheusAnnotations {
-		// Set default Prometheus annotations
-		annotations["prometheus.io/scrape"] = "true"
-		annotations["prometheus.io/port"] = "8888"
-		annotations["prometheus.io/path"] = "/metrics"
-	}
-
-	// allow override of prometheus annotations
```
Comment (swiatekm) [inline internal/manifests/manifestutils/annotations.go:38]: Adding the ConfigMap Hash annotation should also be moved to `PodAnnotations`. We don't need that annotation anywhere else, and this should also reduce the amount of test changes.

Comment (chinaran) [reply internal/manifests/manifestutils/annotations.go:38]: I somewhat agree with you about adding opentelemetry-operator-config/sha256 annotation only to PodAnnotations.
@jaronoff97 @swiatekm 
 can we agree on that?

**Review by iblancasa — APPROVED** [review]

File: internal/manifests/manifestutils/annotations.go (hunk 2 above)
Comment (jaronoff97) [reply internal/manifests/manifestutils/annotations.go:38]: yes, i think that makes sense.

Comment (chinaran) [reply internal/manifests/manifestutils/annotations.go:38]: @swiatekm I've changed it, please check again.

**Review by swiatekm — APPROVED** [review]

**Comment by jaronoff97** [comment]
@chinaran thank you for your contribution!

**Comment by chinaran** [comment]
> @chinaran thank you for your contribution!

Glad to be part of it!

**Pull request merged**